### PR TITLE
CNF-14440: scorecard tests failing

### PR DESCRIPTION
**Pull request by irinamihai**
Description:
- add spec descriptor
- add resources to CRDs
- rename ORANO2IMS to Inventory to make bundle building happy
- adjust the groups and domains to match with the GVK
- upversion to 4.17

### CODE REVIEW

File: api/v1alpha1/inventory_types.go [hunk 1]
```diff
@@ -71,38 +75,43 @@ type AlarmSubscriptionServerConfig struct {
...
+	//+operator-sdk:csv:customresourcedefinitions:type=spec,displayName="IngressHost",xDescriptors={"urn:alm:descriptor:com.tectonic.ui:text"}
 	IngressHost string `json:"ingressHost,omitempty"`
 }
 
 type DeploymentsStatus struct {
-	// +operator-sdk:csv:customresourcedefinitions:type=status,displayName="Deployment Server Status"
+	// +operator-sdk:csv:customresourcedefinitions:type=status,displayName="DeploymentServerStatus"
 	DeploymentServerStatus string `json:"deploymentServerStatus,omitempty"`
-	// +operator-sdk:csv:customresourcedefinitions:type=status,displayName="Metadata Server Status"
+	// +operator-sdk:csv:customresourcedefinitions:type=status,displayName="MetadataServer tatus"
```
Comment (browsell) [inline api/v1alpha1/inventory_types.go:112]: typo

File: internal/controllers/inventory_controller.go [hunk 2]
```diff
@@ -44,9 +44,9 @@ import (
 	"sigs.k8s.io/controller-runtime/pkg/predicate"
 )
 
-//+kubebuilder:rbac:groups=oran.openshift.io,resources=orano2imses,verbs=get;list;watch;create;update;patch;delete
-//+kubebuilder:rbac:groups=oran.openshift.io,resources=orano2imses/status,verbs=get;update;patch
-//+kubebuilder:rbac:groups=oran.openshift.io,resources=orano2imses/finalizers,verbs=update
+//+kubebuilder:rbac:groups=oran.openshift.io,resources=Inventoryes,verbs=get;list;watch;create;update;patch;delete
```
Comment (alegacy) [inline internal/controllers/inventory_controller.go:47]: The plural of `inventory` should be `inventories`, no?

File: config/rbac/role.yaml [hunk 3]
```diff
@@ -155,7 +155,7 @@ rules:
 - apiGroups:
   - oran.openshift.io
   resources:
-  - clusterrequests
+  - Inventoryes
```
Comment (alegacy) [inline config/rbac/role.yaml:158]: `Inventoryes` -> `Inventories`?

Comment (irinamihai) [reply config/rbac/role.yaml]: Updated in the latest patch. Thank you!

File: api/v1alpha1/inventory_types.go [hunk 4]
```diff
@@ -71,38 +75,43 @@ type AlarmSubscriptionServerConfig struct {
...
 	AlarmSubscriptionServerConfig AlarmSubscriptionServerConfig `json:"alarmSubscriptionServerConfig"`
 	// IngressHost defines the FQDN for the IMS endpoints.
 	//+optional
+	//+operator-sdk:csv:customresourcedefinitions:type=spec,displayName="IngressHost",xDescriptors={"urn:alm:descriptor:com.tectonic.ui:text"}
 	IngressHost string `json:"ingressHost,omitempty"`
 }
 
 type DeploymentsStatus struct {
-	// +operator-sdk:csv:customresourcedefinitions:type=status,displayName="Deployment Server Status"
+	// +operator-sdk:csv:customresourcedefinitions:type=status,displayName="DeploymentServerStatus"
```
Comment (alegacy) [inline api/v1alpha1/inventory_types.go:110]: Shouldn't the spaces in the `displayName` be preserved?   Isn't this used in UI elements for end users?

File: internal/controllers/inventory_controller.go [hunk 5]
```diff
@@ -44,9 +44,9 @@ import (
 	"sigs.k8s.io/controller-runtime/pkg/predicate"
 )
 
-//+kubebuilder:rbac:groups=oran.openshift.io,resources=orano2imses,verbs=get;list;watch;create;update;patch;delete
-//+kubebuilder:rbac:groups=oran.openshift.io,resources=orano2imses/status,verbs=get;update;patch
-//+kubebuilder:rbac:groups=oran.openshift.io,resources=orano2imses/finalizers,verbs=update
+//+kubebuilder:rbac:groups=oran.openshift.io,resources=Inventories,verbs=get;list;watch;create;update;patch;delete
+//+kubebuilder:rbac:groups=oran.openshift.io,resources=Inventories/status,verbs=get;update;patch
+//+kubebuilder:rbac:groups=oran.openshift.io,resources=Inventories/finalizers,verbs=update
```
Comment (Missxiaoguo) [inline internal/controllers/inventory_controller.go:49]: Should this be "inventories"?

File: api/v1alpha1/inventory_types.go (hunk 4 above)
Comment (irinamihai) [reply api/v1alpha1/inventory_types.go:110]: I used [TALM](https://github.com/openshift-kni/cluster-group-upgrades-operator/blob/main/pkg/api/clustergroupupgrades/v1alpha1/types.go) as an example which uses both ways, so I've decided to go with no spaces. I can change all of them to include spaces.

**Pull request merged**